packages/@aws-cdk/integ-runner/lib/runner/integration-tests.ts

```diff
@@ -146,6 +146,32 @@ export interface IntegrationTestFileConfig {
  * Discover integration tests
  */
 export class IntegrationTests {
+
+  public static stripPrefixAndSuffix(fileName: string): string {
+    const [suffix, prefix] = Array.from(this.acceptedSuffixPrefixMapping)
+      .find(([suff]) => suff.test(fileName)) ?? ['', ''];
+
+    return fileName.replace(prefix, '').replace(suffix, '');
+  }
+
+  /**
+   * Will discover integration tests with naming conventions typical to each language. Examples:
+   * - TypeScript/JavaScript: integ.test.ts or integ-test.ts or integ_test.ts
+   * - Python/Go: integ_test.py
+   * - Java/C#: IntegTest.cs
+   * @private
+   */
+  private static acceptedSuffixPrefixMapping: Map<RegExp, RegExp> = new Map<RegExp, RegExp>([
+    [new RegExp('\\.(cs|java)$'), new RegExp('^Integ')],
+    [new RegExp('\\.(py|go)$'), new RegExp('^integ_')],
+    [new RegExp('\\.(js)$'), new RegExp('^integ[-_.]')],
```

Comment:
This doesn't appear to support `.ts` files yet?!

Also, when I did my experiments I noticed that we should make sure `.d.ts` files are excluded, as they would cause issues when the TS code is compiled to JS in the same directory.

Reply:
`.ts` files are compiled into `.js` source code, so when executing the tests we actually execute JavaScript files even if they are originally written in TypeScript.
This is backwards compatible with the existing IntegTests that are written in TypeScript, so seems to work (since they pass).

Since `.ts` is not in the list of accepted suffixes, `.d.ts` won't be picked up.

Reply:
Ah I see. I think one important addition will be native TypeScript support using `ts-node` without the need to compile to JS first. This can be modeled as a separate `--language`.

`--app` for TS would be `ts-node {filePath}`